src/app/(admin)/admin/_hooks/common.js

```diff
@@ -0,0 +1,16 @@
+import { useGetList } from 'react-admin';
+import { RESOURCES } from '@admin/_lib/consts';
+
+export const useFindMatchingEntities = ({ key, value }) => {
+  const { data: specialistData } = useGetList(RESOURCES.specialist);
+  const { data: organizationData } = useGetList(RESOURCES.organization);
```

Comment:
Doesn't react-admin provide you a way to filter result instead of getting all? And can you skip a request if no phone number/email provided yet?

Reply:
Refactor the hook so now it should work more efifciently